### ISSUE TITLE
Files being overwritten when already exists in allowed quality

### ISSUE THREAD

**Issue by DrSpaldo** [SickRage/SickRage#3396]
Branch/Commit: master / 826b084110bdef33f9a519686286a5d6f02e1bcc 
OS: Synology
What you did: Shows were already downloaded in 720p BluRay & 720p HDTV
What happened: The same show appeared and was downloaded in 720p BluRay overwriting the old files.
What you expected: That it would not download as I had already met my preferred quality.

To try and prevent it from happening, I made changes to this:

Allowed: 720p HDTV, 720p WEB-DL, 720p BluRay
Preferred: 8K UHD BluRay

As I will never have 8K on my site, I assumed this would prevent this from occurring again.

Unfortunately, it has not. It still downloads & replaces the files when a Bluray version appears, even though I already have the files downloaded on my server. I hope this makes sense :)

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by sharkykh** [comment]
Please include debug logs

**Comment by DrSpaldo** [comment]
```
2017-03-14 13:43:00 DEBUG    POSTPROCESSOR-AUTO :: Processing folder /mnt/whatbox/Six/SIX.S01.720p.BluRay.x264-DEMAND [no rar]
2017-03-14 13:43:00 DEBUG    POSTPROCESSOR-AUTO :: Parsed six.part.1.720p.bluray.x264-demand.mkv into six - E01 [GROUP: demand] [ABD: False] [ANIME: False] [whichReg: [u'no_season_general']]
2017-03-14 13:43:00 DEBUG    POSTPROCESSOR-AUTO :: Trying to find six s on theTVDB
2017-03-14 13:43:00 INFO     POSTPROCESSOR-AUTO :: Processing /mnt/whatbox/Six/SIX.S01.720p.BluRay.x264-DEMAND [no rar]/six.part.1.720p.bluray.x264-demand.mkv (None)
2017-03-14 13:43:00 DEBUG    POSTPROCESSOR-AUTO :: Analyzing name six.part.1.720p.bluray.x264-demand.mkv
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parsed six.part.1.720p.bluray.x264-demand into six - E01 [GROUP: demand] [ABD: False] [ANIME: False] [whichReg: [u'no_season_general']]
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parse result not sufficient (all following have to be set). will not save release name
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parse result(series_name): six
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parse result(season_number): None
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parse result(episode_numbers): [1]
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO ::  or Parse result(air_date): None
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Parse result(release_group): demand
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Don't have a season number, but this show appears to only have 1 season, setting season number to 1...
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Retrieving episode object for 1x1
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Snatch history had a quality in it, using that: 720p BluRay
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Quality of the episode we're processing: 720p BluRay
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: This was manually downloaded but it appears to be better quality than what we have so I'm marking it as priority
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Is ep a priority download: True
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: File /volume3/TV/Six/Season 1/Six - S01E01.mkv is larger than /mnt/whatbox/Six/SIX.S01.720p.BluRay.x264-DEMAND [no rar]/six.part.1.720p.bluray.x264-demand.mkv
2017-03-14 13:43:01 INFO     POSTPROCESSOR-AUTO :: This download is marked a priority download so I'm going to replace an existing file if I find one
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Trying to determine free space on destination drive
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: No associated files for /volume3/TV/Six/Season 1/Six - S01E01.mkv were found during this pass
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Deleting file /volume3/TV/Six/Season 1/Six - S01E01.mkv
2017-03-14 13:43:01 INFO     POSTPROCESSOR-AUTO :: Trying to clean any empty folders under /volume3/TV/Six/Season 1
2017-03-14 13:43:01 DEBUG    POSTPROCESSOR-AUTO :: Using original release name six.part.1.720p.bluray.x264-demand.mkv
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Folder name (SIX.S01.720p.BluRay.x264-DEMAND [no rar]) appears to be a valid release name. Using it.
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Parsed /volume3/TV/Six/Season 1/Six - S01E01 into Six - S01E01 [ABD: False] [ANIME: False] [whichReg: [u'standard', u'season_only']]
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Using cached parse result for: six.part.1.720p.bluray.x264-demand
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Found codec for 'SIX: Pilot'.
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: formatting pattern: %SN - S%0SE%0E -> SIX - S01E01
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Parsed /volume3/TV/Six/Season 1/Six - S01E01 into Six - S01E01 [ABD: False] [ANIME: False] [whichReg: [u'standard', u'season_only']]
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Using cached parse result for: six.part.1.720p.bluray.x264-demand
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Found codec for 'SIX: Pilot'.
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: formatting pattern: Season %S -> Season 1
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Destination folder for this episode: /volume3/TV/Six/Season 1
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Checking if the path /volume3/TV/Six/Season 1 already exists
2017-03-14 13:43:02 DEBUG    POSTPROCESSOR-AUTO :: Copying file from /mnt/whatbox/Six/SIX.S01.720p.BluRay.x264-DEMAND [no rar]/six.part.1.720p.bluray.x264-demand.mkv to /volume3/TV/Six/Season 1/SIX - S01E01.mkv
2017-03-14 13:50:04 DEBUG    POSTPROCESSOR :: Starting new thread: POSTPROCESSOR
2017-03-14 13:50:04 INFO     POSTPROCESSOR :: /mnt/whatbox is already being processed right now, please wait until it completes before trying again
2017-03-14 14:00:04 DEBUG    POSTPROCESSOR :: Starting new thread: POSTPROCESSOR
2017-03-14 14:00:04 INFO     POSTPROCESSOR :: /mnt/whatbox is already being processed right now, please wait until it completes before trying again
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Setter sets location to /volume3/TV/Six/Season 1/SIX - S01E01.mkv
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Transaction with 2 queries executed
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Parsed /volume3/TV/Six/Season 1/SIX - S01E01 into SIX - S01E01 [ABD: False] [ANIME: False] [whichReg: [u'standard', u'season_only']]
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Using cached parse result for: six.part.1.720p.bluray.x264-demand
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Found codec for 'SIX: Pilot'.
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: formatting pattern: %SN - %Sx%0E - %EN - %QN -> SIX - 1x01 - Pilot - 720p BluRay
2017-03-14 14:02:31 DEBUG    POSTPROCESSOR-AUTO :: Notification for NMJ scan update not enabled, skipping this notification
2017-03-14 14:02:31 INFO     POSTPROCESSOR-AUTO :: Processing succeeded for /mnt/whatbox/Six/SIX.S01.720p.BluRay.x264-DEMAND [no rar]/six.part.1.720p.bluray.x264-demand.mkv
```

**Comment by sharkykh** [comment]
If you want to download only one 720p from HDTV or BluRay, you don't need to put anything in preferred qualities, select them **only** in allowed qualities. This is the way it has worked for me so far without any issues.

Sorry I wasn't specific -
In order to help you I need the snatch debug logs - when a search is performed and a download link is sent to the client, not post processing.

**Comment by sharkykh** [comment]
SickRage doesn't support multiple episode qualities, **only one**.
You can choose to download the first match of 720p from HDTV or BluRay, by selecting them **only** in allowed qualities. If you want SickRage to keep looking for another quality **to replace** the existing quality (downloaded according to 'allowed qualities') that is what Preferred Quality is for.

**Comment by DrSpaldo** [comment]
Hi, perhaps I wasn't clear enough. I only want one copy of each episode.  

What I mean is that I have HDTV/WebDL/Bluray set as allowed. So I only want one copy of the file then for it to stop. However, say I have a HDTV copy already, then the Bluray copy comes it, it downloads it a second time and overwrites the Bluray. It is a waste of download/time.

I tried setting 8K as preferred so it never overwrote it, but, that didn't help either.

Let's say I only selected HDTV and there was never a HDTV version released, only WebDL, then it wouldn't work. I don't mind getting rid of bluray, but, you really need both of those versions at the least..

**Comment by DrSpaldo** [comment]
Hi sharkykh, I am not sure where your reply went, I can't see it in here but I can see it in the email.

The reason I have posted the post processing logs & not snatch is that is how I get my files. I have been told that the program isn't designed exactly for that method and I am in the minority of how people use SR. However, that is how I have to use it for my setup. 

This takes us a little off track, but, I would rather you understand. I have a shared seedbox and it automatically gets all the shows everyone watches, so it is not just limited to mine. It downloads & extracts them into directories.

So to avoid it interfering with other peoples files and rutorrent, I set SR to do copy post processing to find the files as I have mounted the remote seedbox via ssh on my synology. Hope that makes sense...

**Comment by sharkykh** [comment]
In my mind, this is not a bug.
What you're asking is that post processing will follow the qualities set in the show's settings.
If we do that, that would cause problems for other people - manually downloaded episodea

**Comment by DrSpaldo** [comment]
I am just suggesting/asking/thinking that if the quality is already met and you have the file. Why would post processing download it / replace it?

**Comment by sharkykh** [comment]
because it's being marked as a priority download
`This was manually downloaded but it appears to be better quality than what we have so I'm marking it as priority`
I don't have enough information to know why.
All I know is, [SR believes it is in a higher quality](https://github.com/SickRage/SickRage/blob/8b368875c395b3279e0e68248e9e1c5f2c351828/sickbeard/postProcessor.py#L923-L926) because the current quality of the episode in your database is lower than the new one.

This **might be** the cause of you trying to have two qualities of one episode.
Like I said before, SickRage doesn't support multiple qualities so it only keeps the status & quality of the one.
In this example, I'm guessing the HDTV was saved - that's why the BluRay was processed again.

**Comment by DrSpaldo** [comment]
Thanks for the reply sharkykh, I think you are right. The downloaded one was HDTV and a Bluray one popped up in my Seedbox directory. At the time I had HDTV and Bluray as "allowed" and 8K set as preferred. 

I would just assume if the allowed quality was met that it wouldn't be overwritten or downloaded.

In any event; I guess my way of thinking is different to the developers and I might have the remove Bluray... this isn't a big deal, but, if I have WebDL and HDTV then I might get this happening again. 

Let's say I was happy to have either WebDL or HDTV and I didn't care as long as one downloaded, now would that be set?

**Comment by sharkykh** [comment]
>Let's say I was happy to have either WebDL or HDTV and I didn't care as long as one downloaded, now would that be set?

**TL;DR:** Allowed Qualities: HDTV + Web-DL
SickRage will **snatch** the best quality of the two it can find - **only once**.

-----
Whatever you set in "Allowed Qualities" will be downloaded **once** - the highest quality the (Manual/Daily/Backlog) search can find.
If you want SR to download a better quality when it's available - set the "better" quality in "Preferred Qualities".

I remind you, these settings affect only the episodes SickRage searches and snatches - not anything that pops up in your post-processing dir.

**Whatever** shows up in the post-processing dir will be processed as manually downloaded **if it wasn't snatched by SickRage**.
Meaning (if I got the way your seedbox works),
if someone else downloads a **better quality** than the current quality you have
**it will usually overwrite** your quality automatically.
